pkg/automationconfig/automation_config_test.go

```diff
@@ -176,55 +176,16 @@ func TestVersionManifest_BuildsForVersion(t *testing.T) {
 	assert.Empty(t, version.Builds)
 }
 
-func TestTLS(t *testing.T) {
-	caPath := "/path/to/ca"
-	certAndKeyPath := "/path/to/cert"
-	mode := TLSModeRequired
+func TestModifications(t *testing.T) {
+	incrementVersion := func(config *AutomationConfig) {
+		config.Version += 1
+	}
 
 	ac, err := NewBuilder().
-		SetName("my-rs").
-		SetDomain("my-ns.svc.cluster.local").
-		SetMongoDBVersion("4.2.0").
-		SetMembers(3).
-		SetFCV("4.0").
-		SetTLS(caPath, certAndKeyPath, mode).
+		AddModifications(incrementVersion, incrementVersion, incrementVersion).
+		AddModifications(NOOP()).
 		Build()
 
 	assert.NoError(t, err)
-	assert.Len(t, ac.Processes, 3)
-
-	// Ensure every process has TLS configured
-	for _, p := range ac.Processes {
-		assert.Equal(t, mode, p.Args26.Net.TLS.Mode)
-		assert.Equal(t, caPath, p.Args26.Net.TLS.CAFile)
-		assert.Equal(t, certAndKeyPath, p.Args26.Net.TLS.PEMKeyFile)
-		assert.True(t, p.Args26.Net.TLS.AllowConnectionsWithoutCertificate)
-	}
-
-	// Ensure the CA was configured as trusted for the agent
-	assert.Equal(t, caPath, ac.TLS.CAFilePath)
-	assert.Equal(t, ClientCertificateModeOptional, ac.TLS.ClientCertificateMode)
```

Comment:
unless I'm missing something, I don't see the corresponding  tests added using the new functional approach. Do we have tests ensuring these fields are set correctly?

Reply:
These were already tested since before in `replicaset_controller_test.go` with `TestAutomationConfig_IsCorrectlyConfiguredWithTLS`.